src/watch_filter.rs

```diff
@@ -76,19 +86,24 @@ impl WatchFilter {
         })
     }
 
-    fn matches_one_of_ignore_patterns(&self, event: &Event) -> bool {
+    fn matches_one_of_watched_patterns(&self, event: &Event) -> bool {
         event.paths().any(|(path, _)| {
-            self.ignore_patterns
-                .iter()
-                .any(|pattern| pattern.matches_path(path))
+            self.config.manifest_pattern.pattern.matches_path(path)
+                || self
+                    .config
+                    .source_patterns
+                    .iter()
+                    .chain(self.config.artifact_patterns.iter())
+                    .any(|wp| wp.active && wp.pattern.matches_path(path))
```

Comment:
This checks `wp.active` whereas the earlier `matches_*_pattern` functions didn't.  Is this intentional?  If so, consider a comment on why; if not, maybe refactor so this calls the `matches_*_pattern` functions so they are guaranteed to stay in sync.

Reply:
This is on purpose. Cleared up in code by renaming `active` and adding more comments.